src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -105,213 +86,216 @@ private static void WriteTestDllAndName(TextWriter writer, string dll, string da
-
-                                    if (!string.IsNullOrEmpty(_options.ExcludeInfo))
-                                    {
-                                        exclude = Exclude(categoryNames, _options.ExcludeInfo);
-                                    }
-
-                                    string data = string.Empty;
-
-                                    //If exclude is true then move on to next method as this one has category marked as exluded
-                                    if (exclude)
-                                    {
-                                        continue;
-                                    }
-                                    else if (include)
-                                    {
-                                        data = FormattedTest(level, type, methodInfo, testCase, data);
-
-                                        //Prevent duplicate entries from being written
-                                        if (!string.IsNullOrEmpty(data) && !testsWritten.Contains(data))
-                                        {
-                                            WriteTestDllAndName(writer, dll, data);
-                                            testsWritten.Add(data);
-                                        }
-                                    }
-                                }
-                            }
-                            //reset categoryNames for next method
-                            categoryNames = new List<string>();
-                        }
-                    }
+                foreach (Type testSuite in testSuites)
+                {
+                    getValidTestsFromTestSuite(testSuite, level);
                 }
             }
-            else
-            {
-                throw new ArgumentNullException("dlls", "no dlls were specified");
-            }
         }
 
         /// <summary>
-        /// Returns the properly formatted test string depending on the level specified. 
+        /// Get all valid test suites from the assembly
         /// </summary>
-        /// <param name="level">Used in the switch statement to determine how much info to put into the data string</param>
-        /// <param name="type">Used to get the namespace of the test</param>
-        /// <param name="methodInfo">Used to get the test info such as declaring type and name</param>
-        /// <param name="testCase">Used to get arguments of a testcase</param>
-        /// <param name="data">The string which is modified and then returned</param>
-        /// <returns>Returns properly formatted test string</returns>
-        private static string FormattedTest(Level level, Type type, MethodInfo methodInfo, TestCaseAttribute testCase, string data)
+        /// <param name="assemblyFullPath">The full qualified path for the assembly</param>
+        /// <returns>Return a list of test suites</returns>
+        private static List<Type> getValidTestSuites(string assemblyFullPath)
```

Comment:
[Should] Our coding convention here is to use Pascal case for both public & private functions.  So GetValidTestSuites instead of getValidTestSuites.  Same with other functions below.

Reply:
OK